Cilsil.Test/Assets/Utils.cs

```diff
@@ -224,6 +224,14 @@ public static string GetString(InferError error)
         public static string Assign(VarName name, string value) =>
             $"{GetString(name)} = {value};\n";
 
+        /// <summary>
+        /// Generates a string representation of a variable null check.
+        /// </summary>
+        /// <param name="name">The name of the variable that has null check.</param>
+        /// <returns>String representation of the null check statement.</returns>
+        public static string NullCheck(VarName name) =>
```

Comment:
It's very hard to understand what the actual code you're testing in when you introduce these kinds of methods (here and also the InitBlock method). Rather than doing this, it's better to add a TestClass method that has the constructs you're trying to examine. For example:

https://github.com/xi-liu-ds/infersharp/blob/5b6caacd9122bf58f16e9b3b50340600b935a439/Cilsil.Test/Assets/TestClass.cs#L185

https://github.com/xi-liu-ds/infersharp/blob/5b6caacd9122bf58f16e9b3b50340600b935a439/Cilsil.Test/Assets/TestClass.cs#L228

Reply:
> It's very hard to understand what the actual code you're testing in when you introduce these kinds of methods (here and also the InitBlock method). Rather than doing this, it's better to add a TestClass method that has the constructs you're trying to examine. For example:
> 
> https://github.com/xi-liu-ds/infersharp/blob/5b6caacd9122bf58f16e9b3b50340600b935a439/Cilsil.Test/Assets/TestClass.cs#L185
> 
> https://github.com/xi-liu-ds/infersharp/blob/5b6caacd9122bf58f16e9b3b50340600b935a439/Cilsil.Test/Assets/TestClass.cs#L228

If we create this unit test in `TestClass.cs`, for example:

```cs
public static void DoExcepionHandlingWithNullCheck(bool doNullCheck)
        {
            StreamReader streamReader = null;
            try
            {
                if (doNullCheck)
                {
                   if (streamReader == null)
                   {
                        return;
                   }
                }
                streamReader.GetHashCode();
            }
            catch(System.IO.IOException e)
            {
                Console.WriteLine(e.Message);
            }
            finally
            {
                streamReader.Close();
            }
        }
```
In this way we kind of implant this null dereference in our codebase as **every unit test** will have Infer anaylze all methods in `TestClass.cs`. The reason why Infer will actually think this method has a null dereference forever because it is possible that `doNullCheck` is `false` and null check will not be invoked.